tests/flows/test_inference.py

```diff
@@ -374,3 +376,35 @@ async def test_run_classifier_inference_on_document_missing(
 def test_iterate_batch(data, expected_lengths):
     for batch, expected in zip(list(iterate_batch(data)), expected_lengths):
         assert len(batch) == expected
+
+
+@pytest.mark.parametrize(
+    "input_stems,expected_output",
+    [
+        ([], []),
+        (
+            ["CCLW.executive.12345.6789", "UNFCCC.document.1234.5678"],
+            ["CCLW.executive.12345.6789", "UNFCCC.document.1234.5678"],
+        ),
+        (["Sabin.document.16944.17490", "Sabin.document.16945.17491"], []),
+        (
+            [
+                "CCLW.executive.12345.6789",
+                "Sabin.document.16944.17490",
+                "UNFCCC.document.1234.5678",
+                "Sabin.document.16945.17491",
+            ],
+            ["CCLW.executive.12345.6789", "UNFCCC.document.1234.5678"],
+        ),
+        (["sabin.document.16944.17490", "SABIN.document.16945.17491"], []),
+        (
+            ["SabinIndustries.document.1234.5678", "DocumentSabin.12345.6789"],
+            ["SabinIndustries.document.1234.5678", "DocumentSabin.12345.6789"],
```

Comment:
Note this example that with the current implementation we would not filter out `DocumentSabin`; happy to update the conditional in the list comprehension.